compiler-k2-utils/src/main/kotlin/com/squareup/anvil/compiler/k2/utils/names/ClassIds.kt

```diff
@@ -0,0 +1,167 @@
+    classId("com.squareup.anvil.annotations.compat", "MergeInterfaces")
+
+  /** `com.squareup.anvil.annotations.compat.MergeModules` */
+  public val anvilMergeModules: ClassId =
+    classId("com.squareup.anvil.annotations.compat", "MergeModules")
+
+  /** `dagger.Binds` */
+  public val daggerBinds: ClassId = classId("dagger", "Binds")
+
+  /** `dagger.Component` */
+  public val daggerComponent: ClassId = classId("dagger", "Component")
+
+  /** `dagger.internal.Factory` */
+  public val daggerFactory: ClassId = classId("dagger.internal", "Factory")
+
+  /** `dagger.internal.Provider` */
+  public val daggerProvider: ClassId = classId("dagger.internal", "Provider")
+
+  /** `dagger.Lazy` */
+  public val daggerLazy: ClassId = classId("dagger", "Lazy")
+
+  /** `dagger.Subcomponent` */
+  public val daggerSubcomponent: ClassId = classId("dagger", "Subcomponent")
+}
+
+private fun classId(
+  packageFqName: String,
+  relativeName: String,
+  isLocal: Boolean = false,
+): ClassId = ClassId(
+  packageFqName = FqName(packageFqName),
+  relativeClassName = FqName(relativeName),
+  isLocal = isLocal,
+)
+
+/**
+ * ```
+ *  given: `com.example.SomeClass
+ * output: `com.example.SomeClass_BindingModule`
+ * ```
+ */
+public val ClassId.bindingModuleSibling: ClassId
+  get() = sibling("${shortClassName.asString()}_BindingModule")
+
+/**
+ * ```
+ *  given: `com.example.SomeClass
+ * output: `com.example.SomeClass_Factory`
+ * ```
+ */
+public val ClassId.factorySibling: ClassId
+  get() = sibling("${shortClassName.asString()}_Factory")
+
+public val ClassId.companion: ClassId get() = child("Companion")
+
+/**
+ * If the receiver ClassId is a top-level class, this function returns a new ClassId with the same
+ * package and the given name.
+ *
+ * ```
+ * val topLevel = ClassId.topLevel(FqName("com.example.SomeClass"))
+ *
+ * topLevel.sibling("SiblingClass").asFqNameString() shouldBe "com.example.SiblingClass"
+ * ```
+ *
+ * If the receiver ClassId is a nested class, this function returns a new ClassId
+ * with the same package and the given name, but nested inside the receiver ClassId.
+ *
+ * ```
+ * val innerClass1 = ClassId(
+ *   FqName("com.example"),
+ *   Name.identifier("SomeClass.InnerClass1")
+ * )
+ *
+ * innerClass1.sibling("InnerClass2").asFqNameString() shouldBe "com.example.SomeClass.InnerClass2"
+ * ```
+ */
+public fun ClassId.sibling(nameString: String): ClassId {
+  return parentClassId?.child(nameString)
+    ?: ClassId(packageFqName, Name.identifier(nameString))
+}
+
+/**
+ * alias for [com.squareup.anvil.compiler.testing.nested]
+ *
+ * ```
+ * val kotlinMap = ClassId.fromString("kotlin/collections.Map")
+ *
+ * val entry by kotlinMap.child()
+ *
+ * entry.asFqNameString() shouldBe "kotlin.collections.Map.Entry"
+ * ```
+ * @see ClassId.createNestedClassId
+ */
+public fun ClassId.child(nameString: String): ClassId = nested(nameString)
+
+/**
+ * alias for [com.squareup.anvil.compiler.testing.nested]
```

Comment:
Looks like this may be copy pasta / could use an update

Reply:
Yep...  I'll get this at some point soon.